File: src/com/entlogic/h5/builder/forms/RenameDialog.java
package com.entlogic.h5.builder.forms;

import java.io.File;

import com.entlogic.h5.Application;
import com.entlogic.h5.annotations.BpCommand;
import com.entlogic.h5.data.DataPackage;
import com.entlogic.h5.data.DataRecord;
import com.entlogic.h5.data.DataTable;
import com.entlogic.h5.db.BD_MODULE;
import com.entlogic.h5.db.QueryStatement;
import com.entlogic.h5.logs.Log;
import com.entlogic.h5.securitys.UserSession;
import com.entlogic.h5.utils.FileUtils;

/**
 * 
 * @author John
 *
 */
public class RenameDialog
{
	@BpCommand(registedName = "修改模块名", logged = false)
	public static DataPackage rename(DataRecord parameters, UserSession session)
	{
		DataPackage dpOut = new DataPackage();			

		try
		{
			String userOid =  parameters.getValue("userOid");
			String appName =  parameters.getValue("appName");
			String formName = parameters.getValue("formName");
			String oldModuleName = parameters.getValue("txtOldModuleName");
			String newModuleName = parameters.getValue("txtNewModuleName");
			
			
			String contextRealPath = Application.getParameter("ContextRealPath");
			String oldPath = null;
			String newPaht = null;
			String contentStr = null;
			
			String oid, newOid;
			
			// 如果该模块名
			if (formName.equals("*"))
			{
				oldPath = contextRealPath + "app_modules/" + oldModuleName;
				newPaht = contextRealPath + "app_modules/" +newModuleName;
				
				File from = new File(oldPath);
				File to = new File(newPaht);
				from.renameTo(to);
				
				BD_MODULE.rename(oldModuleName + "_*", newModuleName + "_*", userOid);	
				
				QueryStatement query = new QueryStatement();
				query.SQL = "selsect OID from BD_MODULE where OID like '" + appName + "_%'";
				DataTable dt = query.excute("jdbc/entlogic", null);
				for (int i = 0; i < dt.getSize(); i++)
				{
					oid = dt.getRecord(i).getValue("OID");
					newOid = oid.replace(oldModuleName, newModuleName);
					BD_MODULE.rename(oid, newOid, userOid);
				}
				
				dpOut.addReturn("modulePath", newModuleName + "_" + formName);
				dpOut.addReturn("result", "success");
			}
			else
			{
				oldPath = contextRealPath + "app_modules/" + appName + "/" + oldModuleName + "_D.html";
				newPaht = contextRealPath + "app_modules/" + appName + "/" + newModuleName + "_D.html";
				if (new File(oldPath).exists()) 
				{
					contentStr = FileUtils.readToString(oldPath);
					contentStr = contentStr.replaceAll(oldModuleName, newModuleName );
					FileUtils.writeToFile(contentStr, newPaht);
					FileUtils.deleteFile(new File(oldPath));
				}
				
				oldPath = contextRealPath + "app_modules/" + appName + "/" + oldModuleName + ".html";
				newPaht = contextRealPath + "app_modules/" + appName + "/" + newModuleName + ".html";
				contentStr = FileUtils.readToString(oldPath);
				contentStr = contentStr.replaceAll(oldModuleName, newModuleName );
				FileUtils.writeToFile(contentStr, newPaht);
				FileUtils.deleteFile(new File(oldPath));
				
				oldPath = contextRealPath + "app_modules/" + appName + "/" + oldModuleName + ".css";
				newPaht = contextRealPath + "app_modules/" + appName + "/" + newModuleName + ".css";
				contentStr = FileUtils.readToString(oldPath);
				FileUtils.writeToFile(contentStr, newPaht);
				FileUtils.deleteFile(new File(oldPath));
				
				oldPath = contextRealPath + "app_modules/" + appName + "/" + oldModuleName + ".js";
				newPaht = contextRealPath + "app_modules/" + appName + "/" + newModuleName + ".js";
				contentStr = FileUtils.readToString(oldPath);
				FileUtils.writeToFile(contentStr, newPaht);
				FileUtils.deleteFile(new File(oldPath));
				
				BD_MODULE.rename(appName + "_" + oldModuleName, appName + "_" + newModuleName, userOid);
				
				dpOut.addReturn("modulePath", appName + "_" + newModuleName);
				dpOut.addReturn("result", "success");
			}
			
			return dpOut;
		}
		catch (Exception e)
		{
			Log.error(e);

			try
			{
				dpOut.addReturn("errorMessage",e.getMessage());
				dpOut.addReturn("result", "faild");
				
				return dpOut;
			}
			catch(Exception ex) 
			{		
				return null;
			}			
		}
	}
}

File: WebContent/app_modules/entlogicBuilder/RenameDialog.js
/////////////////////////////////////////////////////////////////////////////////////////////////
//  常量声明  

// 全局应用
var application = parent.application;

// 应用的根目录
var applicationRoot = parent.applicationRoot;

//应用的根目录
var sessionId = parent.sessionId;

// 页面控件容器
var formControls = new Array();


/////////////////////////////////////////////////////////////////////////////////////////////////
// 数据声明    



/////////////////////////////////////////////////////////////////////////////////////////////////
// 组件声明   
var txtOldModuleName = {};
var txtNewModuleName = {};

/////////////////////////////////////////////////////////////////////////////////////////////////
// 内部变量
var appName = parent.appName;
var formName = parent.formName;


/////////////////////////////////////////////////////////////////////////////////////////////////
// 内部函数


/**
 * 组件初始化
 */ 
function initComponents() {
	// 初始化数据组件
	txtOldModuleName = new entlogic_ui_text("txtOldModuleName", formControls);
	txtOldModuleName.setEnabled(false);
	txtNewModuleName =new entlogic_ui_text("txtNewModuleName", formControls);
	
	$("#btnOK").click( btnOK_onClick);
    OK = btnOK_onClick;
};


/////////////////////////////////////////////////////////////////////////////////////////////////
//页面事件响应

/**
 * 父页面打开事件处理
 */ 
function bodyOnload() {
	// 初始化表单组件
    initKeys();
	initComponents();	
	
	// 接受参数
	if (formName == "*") {
		txtOldModuleName.setValue(appName);
	} else {
		txtOldModuleName.setValue(formName);
	}
};

/**
 * 确定按钮单击事件
 */
function btnOK_onClick() {
	var parameters = new entlogic_data_record();
	parameters.addItem("appName", appName);
	parameters.addItem("formName", formName);
	var dataPackage = postBpForm("com.entlogic.h5.builder.forms.RenameDialog", "rename", formControls, parameters);
	var modulePath = dataPackage.getReturn("modulePath");
	dialogCallBack(modulePath);
	closeDialog();
};
